x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/30_semantic_text_inference.yml

```diff
@@ -0,0 +1,551 @@
+              path: inference_field.inference.chunks
+              query:
+                knn:
+                  field: inference_field.inference.chunks.embeddings
+                  query_vector_builder:
+                    text_embedding:
+                      model_id: dense-inference-id
+                      model_text: "you know, for testing"
+
+  # Search with inner hits
+  - do:
+      search:
+        _source: false
+        index: test-dense-index
+        body:
+          query:
+            nested:
+              path: inference_field.inference.chunks
+              inner_hits:
+                _source: false
+                fields: [inference_field.inference.chunks.text]
+              query:
+                knn:
+                  field: inference_field.inference.chunks.embeddings
+                  query_vector_builder:
+                    text_embedding:
+                      model_id: dense-inference-id
+                      model_text: "you know, for testing"
+
+  - match: { hits.total.value: 2 }
+  - match: { hits.total.relation: eq }
+  - match: { hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.total.value: 2 }
+  - exists:  hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.hits.0.fields.inference_field\.inference\.chunks.0.text
+  - exists:  hits.hits.0.inner_hits.inference_field\.inference\.chunks.hits.hits.1.fields.inference_field\.inference\.chunks.0.text
+
+
+---
+"Updating non semantic_text fields does not recalculate embeddings":
+    - do:
+        index:
+          index: test-sparse-index
+          id: doc_1
+          body:
+            inference_field: "inference test"
+            another_inference_field: "another inference test"
+            non_inference_field: "non inference test"
+
+    - do:
+        get:
+          index: test-sparse-index
+          id: doc_1
+
+    - set: { _source.inference_field.inference.chunks.0.embeddings: inference_field_embedding }
+    - set: { _source.another_inference_field.inference.chunks.0.embeddings: another_inference_field_embedding }
+
+    - do:
+        update:
+          index: test-sparse-index
+          id: doc_1
+          body:
+            doc:
+              non_inference_field: "another non inference test"
+
+    - do:
+        get:
+          index: test-sparse-index
+          id: doc_1
+
+    - match:  { _source.inference_field.text: "inference test" }
+    - match:  { _source.inference_field.inference.chunks.0.text: "inference test" }
+    - match:  { _source.inference_field.inference.chunks.0.embeddings: $inference_field_embedding }
+    - match:  { _source.another_inference_field.text: "another inference test" }
+    - match:  { _source.another_inference_field.inference.chunks.0.text: "another inference test" }
+    - match:  { _source.another_inference_field.inference.chunks.0.embeddings: $another_inference_field_embedding }
+    - match:  { _source.non_inference_field: "another non inference test" }
```

Comment:
Since the test sparse inference service always returns the same embeddings, this test doesn't actually verify that the embeddings were not recalculated...

Reply:
I didn't want to overcomplicate the inference test service, but I think it's worthwhile in order to check this. Done in 356ac9f6bc43768bb1024dc09fc2e23ddaaba5d6